#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From ad630efaa9cf5d45a0e356be29f239b92ac34247 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Mon, 4 Nov 2019 16:20:55 +0100
Subject: [PATCH] Allow build with older KWindowsystem versions

Older versions used the widget and not the window id.
---
 src/commands/command.cpp                | 2 +-
 src/kleopatraapplication.cpp            | 2 +-
 src/uiserver/assuanserverconnection.cpp | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/commands/command.cpp b/src/commands/command.cpp
index 24d70614..565a5d2b 100644
--- a/src/commands/command.cpp
+++ b/src/commands/command.cpp
@@ -246,7 +246,7 @@ void Command::applyWindowID(QWidget *w) const
                 w->setParent(pw, w->windowFlags());
             } else {
                 w->setAttribute(Qt::WA_NativeWindow, true);
-                KWindowSystem::setMainWindow(w->windowHandle(), d->parentWId);
+                KWindowSystem::setMainWindow(w, d->parentWId);
             }
         } else {
             w->setParent(d->parentWidgetOrView(), w->windowFlags());
diff --git a/src/kleopatraapplication.cpp b/src/kleopatraapplication.cpp
index e2e38989..45e70f46 100644
--- a/src/kleopatraapplication.cpp
+++ b/src/kleopatraapplication.cpp
@@ -542,7 +542,7 @@ void KleopatraApplication::openConfigDialogWithForeignParent(WId parentWId)
             d->configureDialog->setParent(pw, d->configureDialog->windowFlags());
         } else {
             d->configureDialog->setAttribute(Qt::WA_NativeWindow, true);
-            KWindowSystem::setMainWindow(d->configureDialog->windowHandle(), parentWId);
+            KWindowSystem::setMainWindow(d->configureDialog, parentWId);
         }
     }

diff --git a/src/uiserver/assuanserverconnection.cpp b/src/uiserver/assuanserverconnection.cpp
index 8e16e0fc..6f1ceb8b 100644
--- a/src/uiserver/assuanserverconnection.cpp
+++ b/src/uiserver/assuanserverconnection.cpp
@@ -234,7 +234,7 @@ static void apply_window_id(QWidget *widget, const QString &winIdStr)
         widget->setParent(pw, widget->windowFlags());
     } else {
         widget->setAttribute(Qt::WA_NativeWindow, true);
-        KWindowSystem::setMainWindow(widget->windowHandle(), wid);
+        KWindowSystem::setMainWindow(widget, wid);
     }
 }

--
2.20.1
